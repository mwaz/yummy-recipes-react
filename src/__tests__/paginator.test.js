import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Paginator from '../components/common/paginator'
import toJson, { shallowToJson } from 'enzyme-to-json';

describe('Paginator component', () => {
    const wrapper = shallow(<Paginator />);
    const preventDefault = jest.fn();

    it('renders properly without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    
    it('previous and next button work', () => {
        expect(wrapper.find('#previous').simulate('click'))
        expect(wrapper.find('#next').simulate('click'))
    });

    
});