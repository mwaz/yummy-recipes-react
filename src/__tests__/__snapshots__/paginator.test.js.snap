// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component renders properly without crashing 1`] = `
<Row
  tag="div"
>
  <Col
    md={5}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
     
  </Col>
  <center>
    <div>
      <Pagination
        tag="ul"
      >
        <Button
          bsStyle="info"
          color="secondary"
          id="previous"
          tag="button"
        >
          Previous 
        </Button>
        <div
          style={
            Object {
              "paddingRight": "10px",
            }
          }
        >
           
        </div>
        <Button
          bsStyle="info"
          color="secondary"
          id="next"
          tag="button"
        >
          Next 
        </Button>
      </Pagination>
    </div>
  </center>
</Row>
`;
